lib/testutils/httpmultibin/httpmultibin.go

```diff
@@ -94,6 +94,12 @@ func getWebsocketHandler(echo bool, closePrematurely bool) http.Handler {
 		if err != nil {
 			return
 		}
+		defer func() {
+			err := conn.Close()
+			if err != nil {
+				panic(fmt.Errorf("close WS conn in httpmultibin failed: %w", err))
```

Comment:
Instead of a `panic()`, I think it will be better to slightly refactor `getWebsocketHandler()` to accept a `t testing.TB` parameter, and then you can simply use `require.NoError(t, err)` or something like that.

Reply:
This a good idea :+1:

Reply:
I reverted because we have some cases where we test abnormal closing, so it is expected to get errors there.